### PR TITLE
docs: Add install with proto example.

### PR DESCRIPTION
**Pull request by milesj**
proto (https://moonrepo.dev/proto) is a new toolchain (version) manager that has support for Bun. This adds a proto example to the install page.

### CODE REVIEW

File: docs/installation.md [hunk 1]
```diff
@@ -38,6 +42,8 @@ $ bun upgrade
 
 {% callout %}
 **Homebrew users** — To avoid conflicts with Homebrew, use `brew upgrade bun` instead.
+
+**proto users** - Use `proto install bun --pin` instead.
```
Comment (colinhacks) [inline docs/installation.md:46]: Bit confused here - why is `--pin` here? If I'm reading the docs correctly, re-running `proto install bun` would re-resolve the version and install the latest version right? Pinning a global default seems like an orthogonal concern.

Comment (milesj) [reply docs/installation.md:46]: It's to ensure the latest version is used when the `bun` shim is executed. proto doesn't symlink the latest version as the shim, it detects the version at runtime.

**Comment by milesj** [comment]
Thanks!
